File: Platforms/Stics_soil_temperature/Tempamp.f90
MODULE Tempampmod
   IMPLICIT NONE
CONTAINS
   !%%CyML Model Begin%%
   SUBROUTINE model_tempamp(min_temp, &
                            max_temp, &
                            tempamp)
      IMPLICIT NONE

      REAL, INTENT(IN) :: min_temp
      REAL, INTENT(IN) :: max_temp
      REAL, INTENT(OUT) :: tempamp

      !- Name: Tempamp -Version: 1.0, -Time step: 1
      !- Description:
      !            * Title: Tempamp model
      !            * Authors: STICS
      !            * Reference: doi:http://dx.doi.org/10.1016/j.agrformet.2014.05.002
      !            * Institution: INRAE
      !            * ExtendedDescription: Calculates temperature amplitude
      !- inputs:
      !            * name: min_temp
      !                          ** description : current minimum temperature
      !                          ** inputtype : variable
      !                          ** variablecategory : auxiliary
      !                          ** datatype : DOUBLE
      !                          ** default : 0.0
      !                          ** min : 0.0
      !                          ** max : 500.0
      !                          ** unit : degC
      !                          ** uri :
      !            * name: max_temp
      !                          ** description : current maximum temperature
      !                          ** inputtype : variable
      !                          ** variablecategory : auxiliary
      !                          ** datatype : DOUBLE
      !                          ** default : 0.0
      !                          ** min : 0.0
      !                          ** max : 100.0
      !                          ** unit : degC
      !                          ** uri :
      !- outputs:
      !            * name: tempamp
      !                          ** description : current temperature amplitude
      !                          ** variablecategory : state
      !                          ** datatype : DOUBLE
      !                          ** min : 0.0
      !                          ** max : 500.0
      !                          ** unit : degC
      !                          ** uri :
      !%%CyML Compute Begin%%
      tempamp = max_temp - min_temp
      !%%CyML Compute End%%
   
   END SUBROUTINE model_tempamp
   !%%CyML Model End%%
END MODULE Tempampmod

File: Platforms/Stics_soil_temperature/src/cs/Stics_soil_temperature/Temp_amp.cs
using System;
using System.Collections.Generic;
using System.Linq;
public class temp_amp
{
    
        public temp_amp() { }
    
    public void  CalculateModel(SoiltempState s, SoiltempState s1, SoiltempRate r, SoiltempAuxiliary a, SoiltempExogenous ex)
    {
        //- Name: temp_amp -Version: 1.0, -Time step: 1
        //- Description:
    //            * Title: temp_amp model
    //            * Authors: None
    //            * Reference: doi:http://dx.doi.org/10.1016/j.agrformet.2014.05.002
    //            * Institution: INRAE
    //            * ExtendedDescription: None
    //            * ShortDescription: None
        //- inputs:
    //            * name: min_temp
    //                          ** description : current minimum temperature
    //                          ** inputtype : variable
    //                          ** variablecategory : exogenous
    //                          ** datatype : DOUBLE
    //                          ** max : 50.0
    //                          ** min : -50.0
    //                          ** default : 0.0
    //                          ** unit : degC
    //            * name: max_temp
    //                          ** description : current maximum temperature
    //                          ** inputtype : variable
    //                          ** variablecategory : exogenous
    //                          ** datatype : DOUBLE
    //                          ** max : 50.0
    //                          ** min : -50.0
    //                          ** default : 0.0
    //                          ** unit : degC
        //- outputs:
    //            * name: temp_amp
    //                          ** description : current temperature amplitude
    //                          ** datatype : DOUBLE
    //                          ** variablecategory : state
    //                          ** max : 100.0
    //                          ** min : 0.0
    //                          ** unit : degC
        double min_temp = ex.min_temp;
        double max_temp = ex.max_temp;
        double temp_amp;
        temp_amp = max_temp - min_temp;
        s.temp_amp= temp_amp;
    }
}